src/frontend/flight/sync/writer.rs

```diff
@@ -770,48 +844,62 @@ mod tests {
     const T3: &str = "table_3";
     static A: &str = "origin-A"; // first origin
     static B: &str = "origin-B"; // second origin
-    static FLUSH: (&str, &str, i64) = ("__flush", "-1", -1);
+    static FLUSH: (&str, &str, Option<i64>) = ("__flush", "-1", None);
 
     #[rstest]
     #[case::basic(
-        &[(T1, A, 1), (T2, A, 2), (T1, A, 3), FLUSH, FLUSH],
-        vec![vec![Some(1), Some(3)]]
+        &[(T1, A, Some(100)), (T2, A, Some(200)), (T1, A, Some(300)), FLUSH, FLUSH],
+        vec![vec![Some(100), Some(300)]]
     )]
     #[case::basic_2_origins(
-        &[(T1, A, 1), (T2, B, 1001), (T1, A, 3), FLUSH, FLUSH],
+        &[(T1, A, Some(100)), (T2, B, Some(200)), (T1, A, Some(300)), FLUSH, FLUSH],
         vec![
-            vec![Some(3), Some(3)],
-            vec![None, Some(1001)],
+            vec![Some(100), Some(300)],
+            vec![None, Some(200)],
+        ]
+    )]
+    #[should_panic(
+        expected = "Transaction from a new origin (origin-B) started without finishing the transaction from the previous one (origin-A)"
+    )]
+    #[case::incomplete_transaction(
+        &[(T1, A, None), (T2, B, Some(100)), FLUSH],
+        vec![],
+    )]
```

Comment:
@eatonphil I'm working under the assumption that this isn't possible (hence the error).

Reply:
Perhaps it should be ignored rather than panic? Panicing is fine for tests but while this isn't a "valid" scenario it is one that may actually happen.

Reply:
Oh it's not actually a panic, or to be more specific it is in a test, but in practice it just responds with an Error to the rpc call.